src/30_Advanced/SeatMap.html

```diff
@@ -0,0 +1,224 @@
+      }
+      .content {
+        background: white;
+        flex-grow: 1;
+        display: flex;
+        flex-direction: column;
+        border-radius: 5px;
+      }
+      .disabled {
+        pointer-events: none;
+        opacity: 0.5;
+        cursor: default;
+      }
+      .hidden {
+        display: none;
+      }
+      .clickable {
+        cursor: pointer;
+      }
+      .zoom-button {
+        bottom: 60px;
+        right: 0px;
+        z-index: 1;
+        border-color: Transparent;
+        background-color: Transparent;
+        margin: 8px;
+        width: 40px;
+        height: 40px;
+        padding: 0;
+        border-width: 0;
+        box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.25);
+        background-color: white;
+        border-radius: 3px;
+      }
+      .opaque {
+        opacity: 0.5;
+        filter: alpha(opacity=50);
+      }
+      #choose-seats-button {
+        padding-bottom: 15px;
+        padding-right: 10px;
+      }
+      .seat-map-svg {
+        box-sizing: border-box;
+        padding-top: 30px;
+      }
+      #pan-zoom {
+        display: initial;
+      }
+      #pan-zoom > svg {
+        left: 0;
+        right: 0;
+        margin: auto;
+      }
+      .hide {
+        display: none;
+      }
+      .list-container {
+        position: relative;
+        height: 70vh;
+      }
+      .ampstart-card {
+        min-width: 100%;
+        padding: 1rem;
+      }
+    </style>
+    <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style>
+    <noscript>
+      <style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style>
+    </noscript>
+
+  </head>
+  <body>
+
+    <!-- Here we are using an `amp-state` called `seatmap`.
+      `scale` variable is used to keep track of the scale of the content being zoomed using by `amp-pan-zoom`.
+      `zoom` variable is used to keep track of the seat map is in a zoomed state or not. This variable is used to
+    decide whether to show a + or - button.
+    `selectedSeats` variables keeps a  list of selected seat ids.  -->
+    <amp-state id="seatmap">
+      <script type="application/json">
+        {
+          "scale": 1,
+          "selectedSeats": [],
+          "zoom": 0
+        }
+      </script>
+    </amp-state>
+
+    <h1>Seat map</h1>
+    <!-- You can implement a seat map by using `amp-list`, `amp-bind` and `amp-pan-zoom` where
```

Comment:
Can you please explain in more details what's going on, what are the challenges that you solve how.

Reply:
done